### PR TITLE
core: do not return 'skipped' when Condition*= fail with StartUnitWithFlags()

### PR DESCRIPTION
**Pull request by bluca**
Backward incompatible change to avoid returning 'skipped' if a condition causes
a job activation to be skipped when using StartUnitWithFlags().
Job results are broadcasted, so it is theoretically possible that existing
software could get confused if they see this result.

Replaces https://github.com/systemd/systemd/pull/22369

Note that instead of a full revert, I'm keeping the D-Bus method in place. We might need it in the future for the next iteration of this feature as discussed, so no need to do a full compat break.

### CODE REVIEW

File: src/core/dbus-unit.c
Comment (yuwata) [inline src/core/dbus-unit.c:423]: The flag `BUS_UNIT_QUEUE_RETURN_SKIP_ON_CONDITION_FAIL` is not used now. Maybe it is better to drop it from the enum?

Comment (bluca) [reply src/core/dbus-unit.c:423]: As discussed yesterday I'm going to rework this shortly, so I'd prefer to keep churn to a minimum as this will be needed again for that

Comment (yuwata) [reply src/core/dbus-unit.c:423]: I see, OK.

File: NEWS
Comment (yuwata) [inline NEWS]: Not a matter in this PR, but maybe we should create a new section about the compat breakages.

Comment (bluca) [reply NEWS]: yes, when we adjust the NEWS before release we should definitely do that

**Review by yuwata — APPROVED** [review]
LGTM. But let's wait for @poettering's final review.

File: NEWS
Comment (poettering) [reply NEWS]: please make clear that this reverts to the status quo ante, i.e. unconditionally reverts to semantics JobRemoved had in 249 and before.

**Comment by poettering** [comment]
Could you add a comment next to definition of `BUS_UNIT_QUEUE_RETURN_SKIP_ON_CONDITION_FAIL` saying `FIXME: currently not used, will be changed soon` or so?

File: NEWS
Comment (bluca) [reply NEWS]: done

**Comment by mrc0mmand** [comment]
> @mrc0mmand this is a known one right?
> 
> ```
> 15:05:12 FAILED TASKS:
> 15:05:12 -------------
> 15:05:12 TEST-53-ISSUE-16347
> ```

That's a new one. I'll restart the job just for good measure.

**Comment by bluca** [comment]
I think I've seen it barf on ubuntu. Previous run passed, and the push has only changed NEWS and a comment, so it should be fine.

**Pull request merged**
